File: 2095. Delete the Middle Node of a Linked List/solution.js
/** Definition for singly-linked list. */
function ListNode(val) {
    this.val = val;
    this.next = null;
}

/**
 * @param {ListNode} head
 * @return {boolean}
 */
var deleteMiddle = function(head) {
    if (!head.next) {
        return null;
    }

    let slow = head, fast = head, pre = head;
    while (fast && fast.next) {
        fast = fast.next.next;
        pre = slow;
        slow = slow.next;
    }
    pre.next = slow.next;
    return head;
};

File: 2095. Delete the Middle Node of a Linked List/solution.py
from typing import *

# Definition for singly-linked list.
class ListNode:
    def __init__(self, x):
        self.val = x
        self.next = None

class Solution:
    def deleteMiddle(self, head: Optional[ListNode]) -> bool:
        if not head.next:
            return None
        slow = fast = prev = head
        while fast and fast.next:
            fast = fast.next.next
            prev = slow
            slow = slow.next
        prev.next = slow.next
        return head